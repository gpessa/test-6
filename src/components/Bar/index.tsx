import React from "react";

import styles from "./styles.module.scss";

const Bar: React.FC = ({ children }) => (
  <div className={styles.element}>{children}</div>
);

export default Bar;
