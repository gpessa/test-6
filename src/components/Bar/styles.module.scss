@import "../../theme/variables.scss";

:local(.element) {
  background-color: $white;
  top: 0;
  z-index: 10;
  position: sticky;
  display: flex;
  vertical-align: middle;
  align-items: center;
  flex-wrap: wrap;
  justify-content: space-between;
}
