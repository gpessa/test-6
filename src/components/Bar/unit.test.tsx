import React from "react";
import Bar from "./";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

describe("<Bar>", () => {
  it("Render correctly Bar component", () => {
    const element = renderer
      .create(
        <Bar>
          <h1>Cane</h1>
          <h2>Gatto</h2>
        </Bar>
      )
      .toJSON();

    expect(element).toMatchSnapshot();
  });

  test("Children are render inside the component", () => {
    const component = mount(
      <Bar>
        <h1>Cane</h1>
        <h2>Gatto</h2>
      </Bar>
    );

    expect(component.find("h1").length).toBe(1);
    expect(component.find("h2").length).toBe(1);
  });
});
