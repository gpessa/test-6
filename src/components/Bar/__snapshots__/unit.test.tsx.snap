// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar> Render correctly Bar component 1`] = `
<div
  className="element"
>
  <h1>
    Cane
  </h1>
  <h2>
    Gatto
  </h2>
</div>
`;
